### ISSUE TITLE
Crash Involving CredentialBootstrap.defaultCredential().delete()

### ISSUE THREAD

**Issue by Tyree-McGee** [okta/okta-mobile-kotlin#254]
### Describe the bug?

`CredentialBootstrap.defaultCredential().delete()` is causing the following crash:
```
java.lang.ArrayIndexOutOfBoundsException: length=1; index=-1
  at java.util.ArrayList.remove(ArrayList.java:506)
  at com.okta.authfoundation.credential.SharedPreferencesTokenStorage$remove$2.invoke(SharedPreferencesTokenStorage.java:102)
  at com.okta.authfoundation.credential.SharedPreferencesTokenStorage$remove$2.invoke(SharedPreferencesTokenStorage.java:100)
  at com.okta.authfoundation.credential.SharedPreferencesTokenStorage$accessStorage$2.invokeSuspend(SharedPreferencesTokenStorage.java:126)
  at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(BaseContinuationImpl.java:33)
  at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.java:106)
  at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.java:42)
  at kotlinx.coroutines.scheduling.TaskImpl.run(TaskImpl.java:95)
  at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.java:570)
  at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.java:750)
  at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.java:677)
  at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.java:664)
```
What can cause this exception to occur?

### What is expected to happen?

For the `defaultCredential` to be removed gracefully.

### What is the actual behavior?

On some unknown occasions that we are not able to reproduce, the crash mentioned above occurs.

### Reproduction Steps?

Unable to reproduce.

### SDK Version and Artifact(s) used.

1.1.2

**Comment by satyajitvure91** [comment]
The logout flow that I'm following is 
1. Call webAuthClient.logoutOfBrowser()
2. In onSuccess, do
-  CredentialBootstrap.defaultCredential().delete()
-  CredentialBootstrap.defaultCredential().revokeAllTokens()
Is this supposed to be the correct sequence of operations for logging out?

**Comment by mallikarjun** [comment]
I see the same issue

**Comment by rajdeepnanua-okta** [comment]
This issue should be fixed in #256. I am adding a few more fixes for other bugs and suggestions, and will create a new release with this fix shortly
